gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/ReferralRequestMapper.java

```diff
@@ -65,13 +74,17 @@ public List<ReferralRequest> mapResources(RCMRMT030101UKEhrExtract ehrExtract,
                                               List<Encounter> encounters,
                                               String practiceCode) {
 
-        return mapEhrExtractToFhirResource(ehrExtract, (extract, composition, component) ->
+        var referralRequests = mapEhrExtractToFhirResource(ehrExtract, (extract, composition, component) ->
             extractAllRequestStatements(component)
                 .filter(Objects::nonNull)
                 .filter(this::isNotSelfReferral)
                 .map(requestStatement
-                         -> mapToReferralRequest(ehrExtract, composition, requestStatement, patient, encounters, practiceCode)))
+                    -> mapToReferralRequest(ehrExtract, composition, requestStatement, patient, encounters, practiceCode)))
             .toList();
+
+        populateSupportingInfoWithDocumentReferencesFromLinkSets(ehrExtract, referralRequests);
```

Comment:
I think I'd have a slight preference for the extractions happening before the call to `mapToReferralRequest` and then have that method depend on something. That way all the code populating the `ReferalRequest` object lives in the same cohesive place. It also doesn't _feel_ like it'll be too difficult, but I may be wrong.

Does that make sense?

Reply:
Have refactored so that this now works as you explained.  Made sense to me.